### DISCUSSION TITLE
Tauri application in WinPE

### DISCUSSION THREAD

**Discussion by alexprabhatbara** [tauri-apps/tauri#6372] [General]
I am developed an application to be run in WinPE. But as WinPE does not have Webview2, my Tauri app fails to launch. I need some help to figure out how to launch my Tauri app in WinPE. Please help as this is urgent.

**Comment by FabianLars** [comment]
The .msi is supposed to install the webview. Did you try other install modes as shown here: <https://tauri.app/v1/guides/building/windows#webview2-installation-options>? Especially talking about the `fixedVersion` variant since it doesn't install the webview on the target system.

Or is it not possible to run `.msi` file on WinPE? In this case you _somehow_ have to get webview2 on there without tauri's help. Tauri apps can't work without WebView2 (unless you're using tauri-egui _only_)

**Comment by alexprabhatbara** [comment]
It is not possible to run .msi in WinPE. Also, it is not possible to install any program, portable apps must be used. I tried the fixed version approach without any success. It is also worth mentioning that apps developed using Electron are able to run in WinPE. I am currently migrating an Electron app to Tauri to be run in WinPE and I'm stuck.

**Reply by FabianLars** [reply]
Then i guess this leaves you with these 2 options:
1) Use the fixed runtime version, but bundle it yourself without using tauri's bundler: <https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/distribution#details-about-the-fixed-version-runtime-distribution-mode> - note that this will add ~250mb to the app size, probably making it larger than your old electron app if that's of concern.
2) Stick to electron until Windows adds a WebView2 optional component to winPE/the adk (if ever).

**Comment by aliddell0423** [comment]
Made some progress here. I built in some MSI installation capability into winpe (24H2 26100). I bundled a fixed version of webview2: Microsoft.WebView2.FixedVersionRuntime.133.0.3065.82.x64 with an MSI and was able to install it successfully. However, the Tauri app does not render in out of the box wpe even with webview 2 installed. 

So far I have tried:

- including full directX features from 24H2
- installing the Netfx 4.8 adk package
- installing VC++ 14

![Screenshot 2025-02-28 131151](https://github.com/user-attachments/assets/c8370cd0-a240-4a4d-ac25-dc0057f2f668)

Fairly certain it's to do with webview2 in wpe, which is another beast to tackle.
So far, no soap. Still working on getting it working. Any pointers and ideas would be greatly appreciated.
